command/ssh/certificate.go

```diff
@@ -429,3 +458,27 @@ type oidcPayload struct {
 	jose.Claims
 	Email string `json:"email"` // OIDC indicator
 }
+
+func deriveMachineID() (uuid.UUID, error) {
+	// use /etc/machine-id
+	machineID, err := ioutil.ReadFile("/etc/machine-id")
+	if err != nil {
+		return uuid.Nil, err
+	}
+
+	// 32 bytes, not secret
+	key := []byte("moon machines mortify more minds")
+	mac := hmac.New(sha256.New, key)
+	mac.Write(machineID)
+	machineHash := mac.Sum(nil)
+
+	// convert to uuid, definitely not thread-safe
+	r := bytes.NewReader(machineHash)
+	uuid.SetRand(r)
+	defer uuid.SetRand(nil)
+	u, err := uuid.NewRandom()
```

Comment:
`uuid.FromBytes(machineHash[:16])` is a simpler way to do this.

Reply:
That doesn't make a valid UUID though since the version and type bytes won't be set.

Reply:
ok makes sense.

Reply:
See https://github.com/smallstep/cli/pull/230#discussion_r392529744 for my proposal for a more simple method.

Reply:
We could also just copy the 16 byes and set the version and type byes manually...

Reply:
done